pynamodb/attributes.py

```diff
@@ -288,6 +304,12 @@ def _dynamo_to_python_attr(cls, dynamo_key: str) -> str:
         """
         return cls._dynamo_to_python_attrs.get(dynamo_key, dynamo_key)  # type: ignore
 
+    def _set_discriminator(self) -> None:
+        if self._discriminator:
+            discriminator_attr = self.get_attributes()[self._discriminator]
+            if discriminator_attr.get_discriminator(self.__class__) is not None:  # type: ignore
```

Comment:
ignore what?

Reply:
`get_attributes()` returns `Dict[str, Attribute]`, but `get_discriminator` only exists on `DiscriminatorAttribute`

I'll see if I can type cast `discriminator_attr` to the correct subclass.

Reply:
`cast` will make the intention clearer